src/components/Field/Field.tsx

```diff
@@ -0,0 +1,88 @@
+import styles from "./Field.module.css";
+import Card from "../Card/Card";
+import React from "react";
+import { initialImages } from "../../constants/initialImages";
+import { IOpenCardsItem, IPlayingFieldItem } from "../../utils/types";
+import { shuffleArray } from "../../utils/util";
+
+function Field() {
+  const [openCards, setOpenCards] = React.useState<IOpenCardsItem[]>([]);
+  const [field, setField] = React.useState<IPlayingFieldItem[]>([]);
+  const images = [...initialImages, ...initialImages];
+  shuffleArray(images);
+  React.useEffect(() => {
+    const newField = images.map((image) => {
+      return {
+        url: image,
+        flip: false,
+        off: false,
+      };
+    });
+    setField(newField);
+  }, []);
+  React.useEffect(() => {
+    console.log(openCards);
+    setTimeout(() => {
+      handleCards();
+    }, "1000");
+  }, [openCards]);
+  function handleCards() {
+    if (openCards.length == 2) {
+      if (openCards[0].url === openCards[1].url) {
+        const newField = [...field];
+        newField[openCards[0].index] = {
+          ...field[openCards[0].index],
+          off: true,
+        };
+        newField[openCards[1].index] = {
+          ...field[openCards[1].index],
+          off: true,
+        };
+        setField(newField);
+        setOpenCards([]);
+      } else {
+        const newField = [...field];
+        newField[openCards[0].index] = {
+          ...field[openCards[0].index],
+          flip: false,
+        };
+        newField[openCards[1].index] = {
+          ...field[openCards[1].index],
+          flip: false,
+        };
+        setField(newField);
+        setOpenCards([]);
+      }
+    }
+    console.log(openCards);
+    return;
+  }
+  const handleClick = (index: number, url: string) => {
+    const newField = [...field];
+    newField[index] = { ...field[index], flip: true };
+    setField(newField);
+    if (openCards.length == 0 || openCards[0].index !== index) {
```

Comment:
I'll be annoying about naming, sorry :))

```js
// now I can read var name instead of unwrapping openCards[0].index == index in my head
const isFirstOpenCard = openCards[0].index == index;

if (openCards.length == 0 || !isFirstOpenCard) { ... }
```